### PR TITLE
Add logging messages and fix closing uncomplete handshakes

### PR DESCRIPTION
**Pull request by boaks**
Add logging for complete or failing handshakes.

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
In the meantime one cause of the NPE could be reproduced.
It is the processing of a close notify, when the handshake has a established session, but hasn't completed (sent FINISH, wait for appl. data). The left the handshake in an inconsistent state, which caused the NPE.

Unit tests are currently missing, I hope I can add them the next days.

**Comment by boaks** [comment]
FMPOV, a 2.2.2 should only contain 11a4071 and db8439d

File: scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java [hunk 1]
```diff
@@ -569,13 +571,35 @@ public void sessionEstablished(Handshaker handshaker, DTLSSession session) throw
...
+			}
 			if (ongoingHandshake.compareAndSet(handshaker, null)) {
 				LOGGER.debug("Handshake with [{}] has been completed", handshaker.getPeerAddress());
 			}
 		}
 
 		@Override
 		public void handshakeFailed(Handshaker handshaker, Throwable error) {
+			SerialExecutor executor = serialExecutor;
+			if (executor != null && !executor.isShutdown()) {
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java:593]: should we make the check only available is LOGGER_OWNER is ErrorEnabled ? (same for handshake complete)

Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java:593]: I can add the check, but I would recommend to have it enabled.

**Comment by boaks** [comment]
> It's not clear to me about the other commits should I review it too ?

Depends on the time you're able to spend. The other stuff is mainly moving code and add a unit test.

**Comment by sbernard31** [comment]
I looked at the reamining commit quickly. No objections :)

**Pull request merged**